packages/react-devtools-shared/src/devtools/views/Settings/ComponentsSettings.js

```diff
@@ -83,11 +86,23 @@ export default function ComponentsSettings(_: {}): React.Node {
     [setParseHookNames],
   );
 
+  const [openInEditorURLPreset, setOpenInEditorURLPreset] = useLocalStorage<
+    'vscode' | 'custom',
+  >(LOCAL_STORAGE_OPEN_IN_EDITOR_URL_PRESET, 'vscode');
+
   const [openInEditorURL, setOpenInEditorURL] = useLocalStorage<string>(
     LOCAL_STORAGE_OPEN_IN_EDITOR_URL,
     getDefaultOpenInEditorURL(),
   );
 
+  useEffect(() => {
+    if (openInEditorURLPreset === 'vscode') {
+      setOpenInEditorURL('vscode://file/{path}:{line}');
+    } else {
+      setOpenInEditorURL('custom link');
```

Comment:
Can we leave it empty? Since the option label will be already "Custom"?

Reply:
Also, why are you using useEffect for it? Can we update `editorURL` in `onChange` callback of select?

Reply:
> Also, why are you using useEffect for it? Can we update `editorURL` in `onChange` callback of select?

😅 ah! i was about to refactor dropping the useEffect, something like below
```
   const selectedValue = currentTarget.value;
            setOpenInEditorURLPreset(selectedValue);
            if (selectedValue === 'vscode') {
              setOpenInEditorURL('vscode://file/{path}:{line}');
            } else if (selectedValue === 'custom') {
              setOpenInEditorURL('');
            }
          }}>
```